File: Sort Colors/Sort_colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int start = 0;
        int mid = 0;
        int end = nums.size()-1;
        while(mid<=end){
            if(nums[mid]==0){
                swap(nums[mid],nums[start]);
                start++;
                mid++;
            }else if(nums[mid]==1){
                mid++;
            }else{
                swap(nums[mid],nums[end]);
                end--;
            }

        }



        
    }
};

File: Sort Colors/Sort_colors.java
class Solution {
    public void sortColors(int[] nums) {
        int n = nums.length ; 
        for(int j = 0 ; j<n ; j++){
            for(int i = 0 ; i<n-1 ; i++){
                if(nums[i]>nums[i+1]){
                    int temp = nums[i];
                    nums[i] = nums[i+1];
                    nums[i+1]=temp;
                }
            }
        }
        
    }
}